sql/rpl_gtid_state.cc

```diff
@@ -148,6 +148,21 @@ void Gtid_state::broadcast_owned_sidnos(const THD *thd)
   }
 }
 
+void Gtid_state::get_snapshot_gtid_executed(
+    std::string &snapshot_gtid_executed)
+{
+  global_sid_lock->wrlock();
+  snapshot_gtid_executed.clear();
+  size_t size= executed_gtids.get_string_length() + 1;
+  char * buf= (char *)malloc(size);
```

Comment:
I would go with creating `std::vector<char> buf(size);` here as a temp buffer, then
`executed_gtids.to_string(buf.data());`
and
`snapshot_gtid_executed.assign(buf.begin(), buf.end())`

Reply:
done